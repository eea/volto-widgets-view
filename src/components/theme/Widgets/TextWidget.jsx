import React from 'react';
import cx from 'classnames';

export const TextWidget = ({ children, className }) => (
  <span className={cx(className, 'text', 'widget')}>{children}</span>
);
