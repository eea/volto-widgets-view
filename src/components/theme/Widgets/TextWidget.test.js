import React from "react";
import renderer from "react-test-renderer";
import { TextWidget } from "./TextWidget";

describe("TextWidget", () => {
  it("renders a text view widget component", () => {
    const component = renderer.create(
      <TextWidget
        data="Foo bar"
        schema={{id: "description"}}
        className="metadata"
      />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
