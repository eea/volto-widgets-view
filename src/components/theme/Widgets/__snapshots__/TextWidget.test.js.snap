// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextWidget renders a text view widget component 1`] = `
<span
  className="metadata text widget"
>
  Foo bar
</span>
`;
